src/script/interpreter.cpp

```diff
@@ -1173,8 +1194,20 @@ PrecomputedTransactionData::PrecomputedTransactionData(const CTransaction& txTo)
     hashOutputs = GetOutputsHash(txTo);
 }
 
+
 uint256 SignatureHash(const CScript& scriptCode, const CTransaction& txTo, unsigned int nIn, int nHashType, const CAmount& amount, SigVersion sigversion, const PrecomputedTransactionData* cache)
 {
+    uint32_t nForkSigHashType=nHashType; // see item 10 in https://github.com/Bitcoin-UAHF/spec/blob/master/replay-protected-sighash.md
+    /* Two-way replay protection in BCH was implemented with a quadratic scaling fix replicating a lot of segwit's functionality.  They change the encoded versionbits inside the transaction hash (in addition to their
+       own 'segwit-lite' implementation.  Much/most of the replay-protected modified sighash in BCH is actually implementing that functionality.
+       However, because BTCG has segwit already, that fix is unnecessary.  The only thing necessary is to add the forkid to the nHashType in the hash to change the transaction hash.
```

Comment:
However then most regtests will falling. Maybe we must set the fork height to 0, and do some random fixes.

Reply:
It would be nice if forkid replay protected transactions always use SIGVERSION_WITNESS_V0 signatures even for non-segwit transactions, just like in BCH.  This makes it easier for hardware wallet especially if they already support BCH.

Is there a reason not to use SIGVERSION_WITNESS_V0 signatures for everything (and get rid of the code for old-style signatures)?